packages/tupaia-web/src/features/Map/MapOverlaySelector/MobileMapOverlaySelector.tsx

```diff
@@ -96,19 +103,27 @@ export const MobileMapOverlaySelector = ({
   overlayLibraryOpen,
   toggleOverlayLibrary,
 }: MobileMapOverlaySelectorProps) => {
+  const { projectCode, entityCode } = useParams();
+  const { hasMapOverlays } = useMapOverlays(projectCode, entityCode);
+
   return (
     <Wrapper>
       {!overlayLibraryOpen && (
         <ButtonWrapper>
           <MapOverlayDatePicker />
-          <ExpandButton onClick={toggleOverlayLibrary} aria-controls="overlay-selector">
+          <ExpandButton
+            onClick={hasMapOverlays ? toggleOverlayLibrary : undefined}
```

Comment:
To be really pedantic, we could just disable the button when `hasMapOverlays` is false, which from an accessibility perspective is preferable I think